packages/rpc-core/src/rpc-subscriptions/program-notifications.ts

```diff
@@ -0,0 +1,75 @@
+import { Base58EncodedAddress } from '@solana/addresses';
+
+import {
+    AccountInfoBase,
+    AccountInfoWithBase58Bytes,
+    AccountInfoWithBase58EncodedData,
+    AccountInfoWithBase64EncodedData,
+    AccountInfoWithBase64EncodedZStdCompressedData,
+    AccountInfoWithJsonData,
+    Commitment,
+    RpcResponse,
+    U64UnsafeBeyond2Pow53Minus1,
+} from '../rpc-methods/common';
+
+type ProgramNotificationsMemcmpFilter = Readonly<{
+    offset: U64UnsafeBeyond2Pow53Minus1;
+    bytes: string;
```

Comment:
This type needs to be parametric on `encoding`. When it's `base58` it should be `Base58EncodedBytes` and when `base64` it should be `Base64EncodedBytes`.

You can probably either do this by having a union type that's discriminated on `encoding` and then doing some nonsense in the method signatures, but you might also be able to do it by adding a type parameter to `ProgramNotificationMemcmpFilter` and then doing some `bytes: TEncoding extends 'base64' ? Base64EncodedBytes : Base58EncodedBytes` trickery.

Reply:
I can do this, but we are not doing this lifting for `getProgramAccounts` HTTP methods

Reply:
It's done ✅

Reply:
It's done ✅